src/Var.jl

```diff
@@ -581,6 +583,85 @@ function arecompatible(x::OutputVar, y::OutputVar)
     return (x.dims == y.dims) && (x.dim_attributes == y.dim_attributes)
 end
 
+"""
+    _check_dims_consistent(x::OutputVar, y::OutputVar)
+
+Check the number, type, and unit of dimensions in `x` and `y` are consistent.
+
+If the unit for a dimension is missing, then the unit is not consistent for that dimension. 
+"""
+function _check_dims_consistent(x::OutputVar, y::OutputVar)
+    # Check the number of dimensions is the same
+    x_num_dims = length(x.dims)
+    y_num_dims = length(y.dims)
+    x_num_dims != y_num_dims && error(
+        "Number of dimensions do not match between x ($x_num_dims) and y ($y_num_dims)",
+    )
+
+    # Check whether the types of dimension agree
+    dim_type_x = dim_type.(keys(x.dims))
+    dim_type_y = dim_type.(keys(y.dims))
+    mismatch_dim_type = dim_type_x .!= dim_type_y
+    any(mismatch_dim_type) && error(
+        "Type of dimensions do not agree between x ($dim_type_x) and y ($dim_type_y)",
+    )
+
+    x_dims = collect(keys(x.dims))
+    y_dims = collect(keys(y.dims))
+    x_units = [dim_units(x, dim_name) for dim_name in x_dims]
+    y_units = [dim_units(y, dim_name) for dim_name in y_dims]
+
+    # Check for any missing units 
+    missing_x = (x_units .== "")
+    missing_y = (y_units .== "")
+    (any(missing_x) && any(missing_y)) && error(
+        "Units for dimensions $(x_dims[missing_x]) are missing in x and units for dimensions $(y_dims[missing_y]) are missing in y",
+    )
+    any(missing_x) &&
+        error("Units for dimensions $(x_dims[missing_x]) are missing in x")
+    any(missing_y) &&
+        error("Units for dimensions $(x_dims[missing_y]) are missing in y")
+
+    # Check whether units match between dimensions 
+    not_consistent_units = .!(x_units .== y_units)
+    any(not_consistent_units) && error(
+        "Units for dimensions $(x_dims[not_consistent_units]) in x is not consistent with units for dimensions $(y_dims[not_consistent_units]) in y",
+    )
+    return nothing
+end
+
+"""
+    resampled_as(src_var::OutputVar, dest_var::OutputVar)
+
+Resample `data` in `src_var` to `dims` in `dest_var`.
+
+The resampling performed here is a 1st-order linear resampling. 
+"""
+function resampled_as(src_var::OutputVar, dest_var::OutputVar)
+    _check_dims_consistent(src_var, dest_var)
+
+    src_resampled_data = [
+        src_var.interpolant(pt...) for
+        pt in Iterators.product(collect(values(dest_var.dims))...)
+    ]
```

Comment:
```suggestion
    src_resampled_data = [
        src_var(pt...) for
        pt in Iterators.product(collect(values(dest_var.dims))...)
    ]
```

Is `collect` needed?